contributor-guides/glossary.md

```diff
@@ -7,3 +7,518 @@ Astro components are the basic building blocks of any Astro project. They are HT
+
+-----
+
+## Adaptors
+
+Adaptors are essential for deploying Astro projects to different hosting platforms. Each adaptor is tailored to a specific host, such as Netlify, Vercel, or Cloudflare Pages, ensuring compatibility and streamlined deployment processes. Adaptors handle the nuances of each platform, providing a seamless integration for Astro projects.
+
+For more details, see:
+- [Astro Adaptor API Reference](https://docs.astro.build/en/reference/adapter-reference/)
+ 
+## API
+
+An API (Application Programming Interface) in the context of Astro is a set of tools and protocols used for building and interacting with software applications. Astro's API provides developers with functions, methods, and standards to create dynamic web applications. This includes various components, utilities, and integrations that facilitate interaction between the Astro framework and other services or applications. By leveraging Astro's API, developers can efficiently build, extend, and customize their web projects, tapping into a rich ecosystem of web development functionalities.
+
+For more details, see:
+- [Astro API Reference](https://docs.astro.build/en/reference/api-reference/)
+
+## Build
+
+Building refers to the process of compiling an Astro project into static files. This step transforms components, scripts, and styles into optimized HTML, CSS, and JavaScript, preparing them for deployment. The build process includes essential optimizations like minification and code splitting, crucial for enhancing performance.
+
+For more details, see:
+- [Astro Build Command](https://docs.astro.build/en/reference/cli-reference/#astro-build)
+- [Deploy your Astro Site](https://docs.astro.build/en/guides/deploy/)
+
+## Client Side Rendering
+
+Client Side Rendering involves rendering web components in the browser through JavaScript. Unlike server-side rendering, client side rendering is used for dynamic content that updates in response to user interactions, providing a more interactive and responsive experience.
+
+For more details, see:
+- [Client-Side Scripts](https://docs.astro.build/en/guides/client-side-scripts/#client-side-scripts)
+- [Client Only Directive](https://docs.astro.build/en/reference/directives-reference/#clientonly)
+
+## Cloudflare
+
+Astro's integration with Cloudflare allows for deploying projects on Cloudflare's edge network, enhancing site performance and reliability. This integration benefits from Cloudflare's global distribution, offering faster content delivery to users.
+
+For more details, see:
+- [Deploying on Cloudflare Guide](https://docs.astro.build/en/guides/deploy/cloudflare/)
+
+## Collection Entry
+
+A Collection Entry is a content piece within a collection directory, supporting formats like Markdown, MDX, YAML, and JSON. These entries are crucial for content organization and management projects.
+
+For more details, see:
+- [Content Collections](https://docs.astro.build/en/guides/content-collections/)
+
+## Components
```

Comment:
```suggestion
## Component
```
I will ask others their opinion, but I think the terms should be singular - like a dictionary. LMK what you think of this.